test/framework/src/main/java/org/elasticsearch/test/AbstractQueryVectorBuilderTestCase.java

```diff
@@ -68,12 +70,21 @@ protected T createTestInstance(float[] expected) {
         return createTestInstance();
     }
 
+    protected KnnSearchBuilder parse(XContentParser parser) throws IOException {
+        return KnnSearchBuilder.fromXContent(parser).build(DEFAULT_SIZE);
+    }
+
     public final void testKnnSearchBuilderXContent() throws Exception {
         AbstractXContentTestCase.XContentTester<KnnSearchBuilder> tester = AbstractXContentTestCase.xContentTester(
             this::createParser,
-            () -> new KnnSearchBuilder(randomAlphaOfLength(10), createTestInstance(), 5, 10, randomBoolean() ? null : randomFloat()),
+            () -> new KnnSearchBuilder.Builder().field(randomAlphaOfLength(10))
+                .queryVectorBuilder(createTestInstance())
+                .k(5)
+                .numCandidates(10)
+                .similarity(randomBoolean() ? null : randomFloat())
+                .build(DEFAULT_SIZE),
             getToXContentParams(),
-            KnnSearchBuilder::fromXContent
+            this::parse
```

Comment:
```suggestion
            this::parseKnnSearchBuilder
```
Calling it `parse` is tricky IMO. But, that is a minor thing.